### PR TITLE
Refetch events on new day

### PR DESCRIPTION
**Pull request by quirk0o**
@adamzima @s1mplex @lkudyba @jtkpiotr

### CODE REVIEW

File: client/app/components/conference_room/provider/EventProvider.jsx [hunk 1]
```diff
@@ -16,22 +16,34 @@ export default class EventProvider extends React.Component {
...
     super(...args);
     this.state = { nextEvents: [] };
-    bindAll(this, ['_fetchForToday', 'handleUpdate', '_confirmEvent']);
+    bindAll(this, ['_fetchForToday', 'handleUpdate', '_confirmEvent', 'updateEvents']);
   }
 
   componentDidMount() {
     this._fetchForToday();
     this._channelSubscription = createSubscription(EVENT_CHANNEL, this._fetchForToday);
+    this.setUpdateTimeout();
```
Comment (madejejej) [inline client/app/components/conference_room/provider/EventProvider.jsx]: `updateTimeout` is not a good enough name for a specific end-of-day update

File: client/app/components/conference_room/provider/EventProvider.jsx [hunk 2]
```diff
@@ -16,22 +16,34 @@ export default class EventProvider extends React.Component {
...
   }
 
+  updateEvents() {
+    this._fetchForToday();
+    this.setUpdateTimeout();
+  }
+
+  setUpdateTimeout() {
+    const now = moment();
+    this.updateTimeout = setTimeout(this.updateEvents, now.clone().add(1, 'day').startOf('day') - now);
```
Comment (madejejej) [inline client/app/components/conference_room/provider/EventProvider.jsx]: I would extract `now.clone().add(1, 'day').startOf('day') - now` to variable and name it something like  `timeToBeginningOfTheNextDay`

File: client/test/components/conference_room/provider/EventProvider.spec.jsx [hunk 3]
```diff
@@ -58,4 +62,45 @@ describe('<EventProvider />', () => {
...
+                             component={DummyComponent} {...ownProps} />);
+        EventSource.fetch.reset();
+        clock.tick(1000 * 60 * 60 * 24);
+        expect(EventSource.fetch).to.have.been.calledOnce();
+      });
+    });
+
+    context('on 01:52 PM', () => {
+      let clock;
+      const timeStamp = 1000 * 60 * 52 + 1000 * 60 * 60 * 10;
```
Comment (madejejej) [inline client/test/components/conference_room/provider/EventProvider.spec.jsx]: Parsing dates would be cleaner than manipulating integers ;)

Comment (adamzima) [reply client/test/components/conference_room/provider/EventProvider.spec.jsx]: looks pretty clean to me ^^

File: client/test/components/conference_room/provider/EventProvider.spec.jsx [hunk 4]
```diff
@@ -58,4 +63,46 @@ describe('<EventProvider />', () => {
...
+
+      after(() => {
+        clock.restore();
+      });
+
+      it('reloads page on next day', () => {
+        mount(<EventProvider conferenceRoom={conferenceRoom}
+                             component={DummyComponent} {...ownProps} />);
+        EventSource.fetch.reset();
+        clock.tick(1000 * 60 * 60 * 24);
```
Comment (madejejej) [inline client/test/components/conference_room/provider/EventProvider.spec.jsx]: How about extracting a constant here? `MILLISECONDS_IN_A_DAY`? The same constant is used in line 103

**Pull request merged**